snmp/tests/test_check.py

```diff
@@ -679,7 +679,7 @@ def test_discovery(aggregator):
             aggregator.reset()
     finally:
         check._running = False
-        check._thread.join()
+        del check
```

Comment:
Maybe, if this is correct?

```suggestion
        del check  # This is what the Agent would do when unscheduling the check.
```